source/documentation/managing_apps/autoscaling.erb

```diff
@@ -0,0 +1,288 @@
+
+### Limits
+
+There are two limits which you can apply to an autoscaling policy:
+
+* `instance_max_count` - maximum number of instances to run
+* `instance_min_count` - minimum number of instances to run
+
+The maximum value for `instance_max_count` is determined by your [quota](#quotas).
+
+The minimum value for `instance_min_count` is 1.
+
+### Metrics
+
+You can scale your app based on some provided metrics:
+
+* `memoryused` - how many megabytes of memory an app instance is using
+* `memoryutil` - percentage of memory utilization where `50` is 50%
+* `cpu` - percentage of CPU utilization where `100` is 100% of a single CPU
+* `responsetime` - average time in milliseconds an app instance takes to respond to external HTTP requests
+* `throughput` - number of external HTTP requests processed per second
+
+You can also scale your app based on custom metrics, please refer to the
+[app-autoscaler user guide](https://github.com/cloudfoundry/app-autoscaler/blob/df1c9a05a00d927b69d2fbf885f32e0b4f228359/docs/Readme.md#auto-scale-your-application-with-custom-metrics)
+on GitHub.
+
+### Example
+
+1. Ensure the autoscaling CLI plugin is installed:
+
+    ```
+    cf install-plugin -r CF-Community app-autoscaler-plugin
+    ```
+
+2. Deploy your app, where `APPNAME` is the name of your application:
+
+    ```
+    cf push APPNAME
+    ```
+
+3. Create an autoscaler service to scale your app:
+
+    ```
+    cf create-service autoscaler autoscaler-free-plan scale-APPNAME
+    ```
+
+4. Bind the autoscaler service to your app:
+
+    ```
+    cf bind-service APPNAME scale-APPNAME
+    ```
+
+5. <p>Create an autoscaling policy, the following policy will scale your app:<ul>
+   <li>adding one instance if CPU usage is greater than 50% for 120 seconds</li>
+   <li>removing one instance if CPU usage is less than 50% for 120 seconds</li>
+   </ul></p><p>It will ensure:<ul>
+   <li>there are never more than 5 instances of your app</li>
+   <li>there are never fewer than 3 instances of your app</li>
+   <li>scaling does not occur multiple times within a 60 second period</li>
+   </ul></p><p><!-- this was hard to do, refactorer beware -->
+   Create the following policy in a file called <code>policy.json</code>:
+   <div class="highlight"><pre class="highlight"><code>{
+      "instance_min_count": 3,
+      "instance_max_count": 5,
+      "scaling_rules": [
+        {
+          "metric_type": "cpu",
+          "breach_duration_secs": 120,
+          "threshold": 50,
+          "operator": "<",
+          "cool_down_secs": 60,
+          "adjustment": "-1"
+        },
+        {
+          "metric_type": "cpu",
+          "breach_duration_secs": 120,
+          "threshold": 50,
+          "operator": ">=",
+          "cool_down_secs": 60,
+          "adjustment": "+1"
+        }
+      ]
+   }
+   </code></pre></div></p>
+
+6. Apply the autoscaling policy to your app:
+
+    ```
+    cf apply-autoscaling-policy APPNAME policy.json
```

Comment:
according to the plugin this should be ` cf attach-autoscaling-policy APP_NAME PATH_TO_FILE`

Reply:
In step 5 of the guide we write

>Create the following policy in a file called policy.json

Reply:
There doesn't seem to be an `apply-autoscaling-policy` command

```
INSTALLED PLUGIN COMMANDS:
   attach-autoscaling-policy              Attach a scaling policy to an application
   autoscaling-api                        Set or view AutoScaler service API endpoint
   autoscaling-history                    Retrieve the scaling history of an application
   autoscaling-metrics                    Retrieve the metrics of an application
   autoscaling-policy                     Retrieve the scaling policy of an application
   create-autoscaling-credential          Create custom metric credential for an application
   delete-autoscaling-credential          Delete the custom metric credential of an application
   detach-autoscaling-policy              Detach the scaling policy from an application
   conduit                                spawns a temporary application, binds your desired service and creates an ssh tunnel from the application to your local machine enabling communication directly with the remote service.
```

Reply:
Ah I see what you mean, resolved in 2388685e86e2ef7103d9602879c95a61493ee649